### PR TITLE
chore(aws sns) add FIFO Topic support

### PR DESCRIPTION
**Pull request by Oleksiivanov**
## Description

added FIFO Topic support 
tested with SM

closes #
https://github.com/camunda/connectors/issues/1671

### CODE REVIEW

File: connectors-e2e-test/connectors-e2e-test-aws/src/test/java/io/camunda/connector/e2e/AwsSnsTest.java [hunk 1]
```diff
@@ -47,12 +47,15 @@ public class AwsSnsTest extends BaseAwsTest {
       "../../connectors/aws/aws-sns/element-templates/aws-sns-outbound-connector.json";
   private static final String TEST_TOPIC_NAME = "test-sns-topic";
   private static final String TEST_QUEUE_NAME = "test-sqs-sqs-queue";
```
Comment (sbuettner) [inline connectors-e2e-test/connectors-e2e-test-aws/src/test/java/io/camunda/connector/e2e/AwsSnsTest.java:49]: Why do we have SQS related resources here? Are they required?

File: connectors/aws/aws-sns/element-templates/aws-sns-outbound-connector.json [hunk 2]
```diff
@@ -112,6 +112,28 @@
...
     "type" : "String"
+  }, {
+    "id" : "topic.type",
+    "label" : "Queue type",
+    "description" : "Select SNS topic type. Details at <a href=\"https://aws.amazon.com/sns/features/\">AWS SNS Features</a>",
+    "optional" : false,
+    "value" : "standard",
+    "constraints" : {
+      "notEmpty" : true
+    },
```
Comment (sbuettner) [inline connectors/aws/aws-sns/element-templates/aws-sns-outbound-connector.json:123]: Shouldnt we also update the version of the template?

File: connectors/aws/aws-sns/src/main/java/io/camunda/connector/sns/outbound/model/TopicRequestData.java [hunk 3]
```diff
@@ -32,6 +34,36 @@ public class TopicRequestData {
   @Deprecated
   private String region;
 
+  @TemplateProperty(
+      label = "Queue type",
```
Comment (sbuettner) [inline connectors/aws/aws-sns/src/main/java/io/camunda/connector/sns/outbound/model/TopicRequestData.java:38]: Why do we call it queue type?

File: connectors/aws/aws-sns/src/main/java/io/camunda/connector/sns/outbound/model/QueueType.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+ * under one or more contributor license agreements. Licensed under a proprietary license.
+ * See the License.txt file for more information. You may not use this file
+ * except in compliance with the proprietary license.
+ */
+package io.camunda.connector.sns.outbound.model;
+
+public enum QueueType {
+  standard,
+  fifo
+}
```
Comment (sbuettner) [inline connectors/aws/aws-sns/src/main/java/io/camunda/connector/sns/outbound/model/QueueType.java:12]: Shouldnt it be rather TopicType?

File: connectors/aws/aws-sns/src/main/java/io/camunda/connector/sns/outbound/model/TopicRequestData.java [hunk 5]
```diff
@@ -56,6 +88,15 @@ public class TopicRequestData {
       description = "Message attributes metadata")
   private Map<String, SnsMessageAttribute> messageAttributes;
 
+  @AssertTrue
+  public boolean hasValidQueueProperties() {
```
Comment (sbuettner) [inline connectors/aws/aws-sns/src/main/java/io/camunda/connector/sns/outbound/model/TopicRequestData.java:92]: Again lets rephrase it to topic.

File: connectors/aws/aws-sns/element-templates/aws-sns-outbound-connector.json [hunk 6]
```diff
@@ -112,6 +112,28 @@
       "type" : "zeebe:input"
     },
     "type" : "String"
+  }, {
```
Comment (igpetrov) [inline connectors/aws/aws-sns/element-templates/aws-sns-outbound-connector.json:115]: Since there is a new property, should the version be incremented?

**Review by igpetrov — APPROVED** [review]
LGTM, one suggestion

File: connectors/aws/aws-sns/element-templates/aws-sns-outbound-connector.json (hunk 6 above)
Comment (igpetrov) [reply connectors/aws/aws-sns/element-templates/aws-sns-outbound-connector.json:115]: OK

**Review by sbuettner — APPROVED** [review]
